llm_handler.py

```
@@ -214,22 +526,324 @@ async def call_llm_api(query, message_context=None):
+                        function_args.get('num_results', 5)
+                    )
+                    logger.info(f"✅ web_search completed - returned {len(result.split('\n'))} lines of results")
+                elif function_name == "web_fetch":
+                    result = await tool_web_fetch(function_args.get('url'))
+                    logger.info(f"✅ web_fetch completed - fetched {len(result)} characters from URL")
+                elif function_name == "search_and_summarize":
+                    result = await tool_search_and_summarize(
+                        function_args.get('query'),
+                        function_args.get('num_results', 3)
+                    )
+                    logger.info(f"✅ search_and_summarize completed - returned comprehensive results")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)